File: apps/desktop/src/lib/api/documents.ts
/**
 * Document management API
 * Communicates with Rust backend via Tauri IPC
 */

import { invoke } from '@tauri-apps/api/core';
import type { Document, CreateDocumentInput, UpdateDocumentInput } from '@aycd/core';

/**
 * Creates a new document
 */
export async function createDocument(input: CreateDocumentInput): Promise<Document> {
  return await invoke<Document>('create_document', {
    projectPath: input.projectId, // projectId is actually the path in our case
    title: input.title,
    category: input.category,
    subcategory: input.subcategory,
  });
}

/**
 * Reads a document's content
 */
export async function readDocument(documentPath: string): Promise<Document> {
  return await invoke<Document>('read_document', { documentPath });
}

/**
 * Updates a document's content
 */
export async function updateDocument(input: UpdateDocumentInput): Promise<void> {
  // We need the path from the document
  const documentPath = input.id; // For now, using id as path
  await invoke('update_document', {
    documentPath,
    content: input.content,
  });
}

/**
 * Deletes a document
 */
export async function deleteDocument(documentPath: string): Promise<void> {
  await invoke('delete_document', { documentPath });
}

/**
 * Lists all documents in a specific directory
 */
export async function listDocumentsInDir(dirPath: string): Promise<Document[]> {
  return await invoke<Document[]>('list_documents_in_dir', { dirPath });
}

/**
 * Lists all documents in a project
 */
export async function listAllDocuments(projectPath: string): Promise<Document[]> {
  return await invoke<Document[]>('list_all_documents', { projectPath });
}

File: apps/desktop/src-tauri/src/commands/documents.rs
use crate::models::Document;
use crate::services::document_service;
use std::path::PathBuf;

/// Creates a new document in the project
#[tauri::command]
pub async fn create_document(
    project_path: String,
    title: String,
    category: String,
    subcategory: Option<String>,
) -> Result<Document, String> {
    let path = PathBuf::from(project_path);
    let subcat = subcategory.as_deref();

    document_service::create_document(&path, &title, &category, subcat)
        .map_err(|e| format!("Failed to create document: {}", e))
}

/// Reads a document's content
#[tauri::command]
pub async fn read_document(document_path: String) -> Result<Document, String> {
    let path = PathBuf::from(document_path);

    document_service::read_document(&path)
        .map_err(|e| format!("Failed to read document: {}", e))
}

/// Updates a document's content
#[tauri::command]
pub async fn update_document(document_path: String, content: String) -> Result<(), String> {
    let path = PathBuf::from(document_path);

    document_service::update_document(&path, &content)
        .map_err(|e| format!("Failed to update document: {}", e))
}

/// Deletes a document
#[tauri::command]
pub async fn delete_document(document_path: String) -> Result<(), String> {
    let path = PathBuf::from(document_path);

    document_service::delete_document(&path)
        .map_err(|e| format!("Failed to delete document: {}", e))
}

/// Lists all documents in a specific directory
#[tauri::command]
pub async fn list_documents_in_dir(dir_path: String) -> Result<Vec<Document>, String> {
    let path = PathBuf::from(dir_path);

    document_service::list_documents_in_dir(&path)
        .map_err(|e| format!("Failed to list documents: {}", e))
}

/// Lists all documents in the project
#[tauri::command]
pub async fn list_all_documents(project_path: String) -> Result<Vec<Document>, String> {
    let path = PathBuf::from(project_path);

    document_service::list_all_documents(&path)
        .map_err(|e| format!("Failed to list all documents: {}", e))
}
